src/primitives/transaction.h

```diff
@@ -261,14 +255,18 @@ class COutPoint
     uint256 hash;
     uint32_t n;
 
-    /* If this flag set, the CTxIn including this COutPoint has a
+    /* If this flag is set, the CTxIn including this COutPoint has a
      * CAssetIssuance object. */
     static const uint32_t OUTPOINT_ISSUANCE_FLAG = (1 << 31);
 
+    /* If this flag is set, the CTxIn including this COutPoint
+     * is a peg-in input. */
+    static const uint32_t OUTPOINT_PEGIN_FLAG = (1 << 30);
```

Comment:
I feel like we've had this discussion in person, and if so I apologize for rehashing it. But why are we reusing an input as a peg-in? Is there some connection between inputs and pegins? Doing it for asset issuance was necessary because we needed a source of entropy. But the outpoint being spent isn't used for the pegin, right?

Reply:
From a conceptual level, you are "spending" the bitcoin output, after extracting the "real" scriptPubKey from the pegin witness and matching it to the p2ch.

More practically speaking, re-using the prevout fields:
1) Makes each transaction with a peg-in input have a unique txid(assuming there are no double-pegins allowed)
2) Puts those values under the signature

Reply:
Ah, because the peg-in data is in the witness. That's fair, although I need to dwell on whether this data should be in the witness or not.

My question was actually more along the lines of "Why not introduce a vector of CTxPegin objects?" But the fact that some pegin data is witness causes this to make sense.